tests/torch_trainer.py

```diff
@@ -0,0 +1,154 @@
+import functools
+
+import torch
+import alpa
+import alpa.torch as atorch
+
+
+"""
+====== New APIs (added to `atorch.*` namespace) ======
+1. atorch.set_mode()
+2. atorch.mode()
+3. atorch.to_format()
+4. atorch.assert_format()
+5. atorch.meta_init()
+6. atorch.functionalize()
+7. atorch.materialize()
+8. atorch.value_and_grad()
+9. atorch.manual_seed()
+10. atorch.enable_dist_for_funcs()
+====== New APIs (added to `atorch.*` namespace) ======
+"""
+
+"""
+atorch vs. torch ?
+
+Current design: all `atorch` usage is fully contained within the trainer code (i.e. this file).
+User code e.g. test_torch_simple.py doesn't need to use `atorch` at all.
+
+This is possible because:
+1. We moved `atorch.optim.adam` to a standalone library (`torchoptim`), detached from Alpa framework.
```

Comment:
I moved `torchoptim` out into https://github.com/yf225/torchoptim, so that it's detached from alpa now. Would love to move that `torchoptim` repo into https://github.com/alpa-projects if it sounds good, since its sole purpose is to make Alpa-PyTorch work well.

Reply:
Sounds great - let me move it back to alpa.torch